ZohoCRMCharts/Classes/ZCRMKPI.swift

```diff
@@ -0,0 +1,513 @@
+		self.addSubview(self.titleView)
+		var constraints: [NSLayoutConstraint] = []
+		constraints += NSLayoutConstraint.constraints(withVisualFormat: "H:|-15-[title]", options: [], metrics: nil, views: ["title": self.titleView])
+		constraints += NSLayoutConstraint.constraints(withVisualFormat: "V:|-15-[title]", options: [], metrics: nil, views: ["title": self.titleView])
+		NSLayoutConstraint.activate(constraints)
+		
+	}
+	
+	/**
+		Updates the title view with the given options.
+	*/
+	private func updateTitleText() {
+		
+		self.titleView.attributedText = NSAttributedString(string: self.title, attributes: [.font : self.renderOptions.titleFont, .foregroundColor: self.renderOptions.titleFontColor ])
+	}
+	
+	
+	/**
+		Sets the view for standard, growth index and basic KPI.
+	*/
+	private func setSimpleKpiView(){
+		
+		self.valueView.translatesAutoresizingMaskIntoConstraints = false
+		self.addSubview(self.valueView)
+		if (self.type != .basic){
+			self.comparedToView.translatesAutoresizingMaskIntoConstraints = false
+			self.comparedToView.textAlignment = .right
+			self.addSubview(self.comparedToView)
+		}
+		
+	}
+	
+	/**
+		Adds the constraints for the simple KPI.
+	*/
+	private func addSimpleKpiConstraints() {
+		
+		var constraints: [NSLayoutConstraint] = []
+		constraints += NSLayoutConstraint.constraints(withVisualFormat: "V:[title]-[value]", options: [], metrics: nil, views: ["title": self.titleView, "value": valueView])
+		if !self.isBasic {
+			constraints += NSLayoutConstraint.constraints(withVisualFormat: "H:|-16-[value(==comparedTo)]-[comparedTo]-15-|", options: [.alignAllBottom], metrics: nil, views: ["value": valueView, "comparedTo": self.comparedToView])
+			constraints += NSLayoutConstraint.constraints(withVisualFormat: "V:[comparedTo]-|", options: [], metrics: nil, views: ["comparedTo": self.comparedToView])
+		} else {
+			constraints += NSLayoutConstraint.constraints(withVisualFormat: "H:|-15-[value]", options: [], metrics: nil, views: ["value": valueView])
+		}
+		NSLayoutConstraint.activate(constraints)
+	}
+	
+	/**
+		Sets the data for standard, growth index and basic KPI.
+	*/
+	private func setSimpleKpiData() {
+		
+		let data: ZCRMKPIRow =  self.data[0]
+		
+		if !self.isBasic {
+			self.updateComparedToViewText(text: data.comparedToLabel + ": " + data.comparedToValue)
+		}
+		if self.isStandard {
+			self.valueView.attributedText = ZCRMKPIUIUtil.shared.getValueTextForStandardKPI(data: data, options: self.renderOptions )
+		} else if self.isGrowthIndex {
+			self.valueView.attributedText = ZCRMKPIUIUtil.shared.getValueTextForGrowthIndexKPI(data: data, options: self.renderOptions)
+		} else if self.isBasic {
+			self.valueView.attributedText = ZCRMKPIUIUtil.shared.getValueTextForBasicKPI(data: data, options: self.renderOptions )
+		}
+	}
+	
+	
+	/**
+		Sets the compared to text for standard, growth index and scorecard KPI.
+	*/
+	private func updateComparedToViewText(text: String?) {
+		
+		var textToFit = ""
```

Comment:
Use String()

Reply:
Both are same. why we shouldn't use that?